internal/chain/engine.go

```diff
@@ -135,6 +140,104 @@ func (e *engineImpl) VerifyShardState(
 	return nil
 }
 
+// VerifyVRF implements Engine, checking the vrf is valid
+func (e *engineImpl) VerifyVRF(
```

Comment:
Why should the check function of VRF be moved from the consensus module to the engine module?

Reply:
because verifyVRF belongs to the logic for block verification which is part of the engine logic. Not limited to consensus only.